app-crypt/tpm2-tools/tpm2-tools-5.2.ebuild

```
@@ -0,0 +1,71 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{8,9} )
+inherit autotools bash-completion-r1 python-single-r1
+
+DESCRIPTION="Tools for the TPM 2.0 TSS"
+HOMEPAGE="https://github.com/tpm2-software/tpm2-tools"
+SRC_URI="https://github.com/tpm2-software/${PN}/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+fapi test"
+
+RDEPEND=">=app-crypt/tpm2-tss-3.0.1[fapi?]
+	dev-libs/openssl:=
+	net-misc/curl
+	sys-libs/efivar:=
+	${PYTHON_DEPS}"
+DEPEND="${RDEPEND}
+	test? (
+		app-crypt/swtpm
+		app-crypt/tpm2-abrmd
+		app-editors/vim-core
+		dev-tcltk/expect
+		dev-util/cmocka
+		dev-python/pyyaml
+	)"
+BDEPEND="virtual/pkgconfig
+	sys-devel/autoconf-archive"
+
+RESTRICT="!test? ( test )"
+REQUIRED_USE=" ${PYTHON_REQUIRED_USE} "
+
+# One of the tests fails without this patch. See
+# https://github.com/tpm2-software/tpm2-tools/issues/2767
+PATCHES=(
+	"${FILESDIR}/${PN}-5.1.1-no-efivar-automagic.patch"
+)
+
+src_prepare() {
+	sed -i \
+	"s/m4_esyscmd_s(\[git describe --tags --always --dirty\])/${PV}/" \
+	"${S}/configure.ac" || die
+	"${S}/scripts/utils/man_to_bashcompletion.sh"
+	default
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_enable fapi) \
+		$(use_enable test unit) \
+		--with-bashcompdir=$(get_bashcompdir) \
+		--enable-hardening
+}
+
+src_install() {
+	default
+
+	mv "${D}/$(get_bashcompdir)/tpm2_completion.bash" \
+	   "${D}/$(get_bashcompdir)/tpm2" || die
+	for B in "${D}"/usr/bin/tpm2_*
+	do
+		TPM2_UTILS="${TPM2_UTILS} $(basename ${B})"
+	done
+	bashcomp_alias tpm2 ${TPM2_UTILS}
```

Few things there
1. `${ED}` here too
2. `B` and `TMP2_UTILS` are missing a `local` declaration
3. just style, but usually we use lowercase naming for local variables, e.g. `b` or `util` instead to be more distinguishable
4. also just style but `for var in ...; do` without do/then on its own line is more common in ebuilds
5. `${B}` is missing quotes while `${D}` inside it could have spaces that get split as separate arguments to basename. I know it's strange but you can do quotes-in-quotes here because `$(...)` is its own context, i.e. `="${xvar} $(command "${yvar}")"` works -- but you could instead use `${var##*/}` to strip up to the last / without calling external basename, quotes also won't matter then.
6. could simplify a bit with `+=`, e.g. `TPM2_UTILS+=" ${var##*/}"`

But these aside, I have my own suggestion to simplify this without a loop if you like it:
```suggestion
	local utils=( "${ED}"/usr/bin/tpm2_* )
	bashcomp_alias tpm2 "${utils[@]##*/}"
```
In case not familiar with these, this will create an array of all paths using the glob, and then expand it as arguments like `econf "${myeconfargs[@]}"` would but will remove `*/` from each elements.

---

I like your version better, so I changed over your your version